### PR TITLE
Feature/iresearch test latest format

### PR DESCRIPTION
**Pull request by Dronplane**
### Scope & Purpose

Update iresearch to the latest upstream version:
 + New fixed-length-dense (and faster!) norm2 feature as replacement for the norm.
 + New features flag format
 + Built-in vpack parsing
 + Segmentation analyzer
 + Collation analyzer

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: 3rdParty/iresearch/cmake/OptimizeForArchitecture.cmake [hunk 1]
```diff
@@ -178,9 +178,7 @@ macro(OFA_AutodetectX86)
          endif(_cpu_model GREATER 2)
       endif(_cpu_family EQUAL 6)
    elseif(_vendor_id STREQUAL "AuthenticAMD")
-      if(_cpu_family EQUAL 25)
-         set(TARGET_ARCHITECTURE "zen3")
-      elseif(_cpu_family EQUAL 23)
+      if(_cpu_family EQUAL 23)
```
Comment (gnusi) [inline 3rdParty/iresearch/cmake/OptimizeForArchitecture.cmake:181]: it seems we have to backport this change to iresearch

File: 3rdParty/iresearch/cmake/OptimizeForArchitecture.cmake [hunk 2]
```diff
@@ -350,9 +348,6 @@ macro(OFA_HandleX86Options)
       list(APPEND _march_flag_list "znver1")
       _skylake()
       list(APPEND _available_vector_units_list "sse4a")
-   elseif(TARGET_ARCHITECTURE STREQUAL "zen3")
```
Comment (gnusi) [inline 3rdParty/iresearch/cmake/OptimizeForArchitecture.cmake:353]: it seems we have to backport this change to iresearch

File: arangod/IResearch/IResearchDocument.h [hunk 3]
```diff
@@ -117,11 +124,27 @@ struct Field {
     return true;
   }
 
-  irs::flags const* _features{&irs::flags::empty_instance()};
+  void setFeatures(AnalyzerPool::AnalyzerFeatures const& poolFeatures) {
+    _features = &poolFeatures;
+    //FIXME: check here datastore version and translate accordingly
+    for (auto const& feature : _features->field_features()) {
```
Comment (gnusi) [inline arangod/IResearch/IResearchDocument.h:130]: I suggest we do it at AnalyzerPool creation, potenitally we can have 2 feature sets, e.g. old and a new one

**Comment by gnusi** [comment]
I suggest we also add collation analyzer

**Review by Dronplane — COMMENTED** [review]
LGTM

**Pull request merged**